[securespider] iP

securespider opened the pull request:
# Dukie

> “The way to get started is to quit talking and begin doing.”  – Walt Disney [(source)](https://blog.hubspot.com/sales/famous-quotes)
 
The app is so **easy to use**, you can even use it while doing a dookie. 


![image](https://media.istockphoto.com/id/1257001864/vector/a-cartoon-illustration-of-a-brown-poo-dookie-or-turd.jpg?s=612x612&w=0&k=20&c=MOpwoydkJGmRuy7FkW1-V0fjX9wCP41MdXUpYycYBtQ=)
 
Dukie frees your mind of having to remember things you need to do. It's,
 
* text-based
* easy to pick up
* ~FAST~ SUPER FAST to use
 
Dukie currently supports the following functions:
 
- [x]  Managing tasks
- [x]  Managing deadlines
- [x]  Managing event details
- [ ]  Search / Filter for any task / deadline / event
- [ ]  Reminders (coming soon)
- [ ]  !help (coming soon)

## Instructions for Installation

### All you need to do is,

1. download it from [here](https://github.com/securespider/ip.git).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉


View a snippet of the `main` method here:
```java
public static void main(String[] args) {
        new Duke(".\\tasks.txt").run();
        }
```

Commit:
Add Gradle support

securespider committed:
Level-1

securespider committed:
Level-2

securespider committed:
Level-3

securespider committed:
Level-4

securespider committed:
TextUiTesting

securespider committed:
Level-5

securespider committed:
Level-6

securespider committed:
Added additional test cases for delete command

securespider committed:
A-Enums

securespider committed:
Modified the toString methods to follow syntax

securespider committed:
Added Exceptions

securespider committed:
Changed commands to include labels based on the expected command starting string

securespider committed:
Level-7

securespider committed:
Level-8

securespider committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Ui.java

securespider committed:
A-Packages

securespider committed:
Reformatted UI methods

securespider committed:
Created Command classes for each command

securespider committed:
Added the Parser in to complete more OOP

zheng-ze commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,41 @@
+        String[] inputArray = input.split(" ", 2);
+        String inputCommand = inputArray[0];
+        switch (inputCommand) {
+            case "bye":
```
Cases for switch statements should not be indented.

securespider replied:
thank you!

zheng-ze commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,41 @@
+
+public class Parser {
+    public static Command parse(String input) throws DukeException {
+        String[] inputArray = input.split(" ", 2);
```
Shouldn't the inputArray be called inputs?

securespider replied:
good catch, thank you!

zheng-ze commented on src/main/java/duke/Command/ListCommand.java:
```diff
@@ -0,0 +1,13 @@
+package duke.Command;
+
+import duke.Storage;
```
Good explicit declaration of imports.

zheng-ze commented on src/main/java/duke/Exceptions/CommandException.java:
```diff
@@ -0,0 +1,27 @@
+        super(cause);
+    }
+
+    public CommandException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
```
Shouldn't enableSuppression and writableStackTrace sound like booleans?

securespider replied:
This are the names for the constructor of exception in the JavaDocs

zheng-ze commented on src/main/java/duke/Exceptions/FIleException.java:
```diff
@@ -0,0 +1,27 @@
+package duke.Exceptions;
+
+public class FIleException extends DukeException {
```
Should this be FileException?

zheng-ze commented on src/main/java/duke/Tasks/Event.java:
```diff
@@ -0,0 +1,36 @@
+    public Event(String description, String by, String to) throws DescriptionException {
+        super(description);
+        this.strBy = by;
+        this.strTo = to;
```
Could potentially try to separate by logical units.

securespider replied:
hihi sorry what do you mean by this?

zheng-ze commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,41 @@
+import duke.Tasks.Task;
+import duke.Tasks.TaskList;
+
+import java.io.*;
```
Should make this import explicit.

zheng-ze commented on src/main/java/duke/Tasks/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Override
+    public String toString() {
+        return "D | " + super.toString() + " | " + this.strBy;
```
Good spacing.

securespider committed:
Update README.md

securespider committed:
Update README.md

securespider committed:
Use-GFMD

sanjevi13 commented on src/main/java/duke/Command/Commands.java:
```diff
@@ -0,0 +1,17 @@
+package duke.Command;
+
+public enum Commands {
```
Nice abstraction!

sanjevi13 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+        new Duke(".\\tasks.txt").run();
+    }
+
+    public void run() {
```
Good abstraction and usage of Command class!

sanjevi13 commented on src/main/java/duke/Tasks/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+        int count = 1;
+        StringBuilder result = new StringBuilder();
+        for (Task task : this) {
+            result.append(count++)
```
Maybe there is a better way you can do this? For example, you can try using String.format, though that would mean you need to change a bit more in your code. Small issue though, good work in general!

sanjevi13 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,68 @@
+        System.out.println("What can I do for you?");
+    }
+
+    public void showLine() {
```
Good naming of methods! I can see that you standardize all the method names in most classes! 😄

sanjevi13 reviewed (commented):
I think you may need to include JavaDocs soon 😄, but in general I feel your code is easy to read and I couldn't spot many coding violations, so I believe that implementing JavaDocs wouldn't be an issue for you. I think the only other small issue I spotted was the indentation of the switch statement, and maybe a few import statements, which someone else has already commented on. I can see that you avoided arrowhead style code, so that is good to see as well. Keep up the good work!

securespider committed:
Added gradle support onto my project

securespider committed:
Merge remote-tracking branch 'origin/add-gradle-support'

securespider committed:
Added gradle support

securespider committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.\tasks.txt
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties

securespider committed:
minor changes to build.gradle

securespider committed:
Added some junit tests

securespider committed:
Added javadocs for some methods and classes

securespider committed:
Change the names to fit convention

securespider committed:
Added find command

securespider committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java

securespider committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Tasks/Event.java
#	src/main/java/duke/Tasks/Task.java
#	src/main/java/duke/Ui.java

securespider committed:
Created files for the GUI

securespider committed:
Added Gui capabilities to Duke

securespider committed:
Update build.gradle

JUnit test cases cannot be done and default main class is currently the Duke class

Let's change the dependencies in the build.gradle file to be able to run junit to run its assertion tests
Let's change the default main class to be the gui.Main class which runs the GUI chatbot instead of on CLI

securespider committed:
added some assertions to the tasks and oop objects

securespider committed:
Update AddDeadlineCommand.java to include assertions

Enforce that description cannot be null or an empty string

Task cannot be created as there will be exception thrown

Let's assert that description cannot be empty

securespider committed:
Removed Commands because it was not used much and changed other files to ensure good code quality

securespider committed:
Merge pull request #1 from securespider/branch-A-CodeQuality

A-CodeQuality

securespider committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/task/Task.java

securespider committed:
Merge branch 'master' into branch-A-Assertions

securespider committed:
Merge pull request #2 from securespider/branch-A-Assertions

A-Assertions

securespider committed:
Merge remote-tracking branch 'origin/master'

securespider committed:
Added Duke logo to show up on initialisation

securespider committed:
Added Duke logo to show up on initialisation

securespider committed:
Added alternative storage options

securespider committed:
Added alternative storage options

securespider committed:
Added the user guide

securespider committed:
Merge branch 'branch-C-FlexibleDataSource'

securespider committed:
Merged the FlexibleDataSource extension to the master branch
Changed the storage to always save on update of tasklist

securespider committed:
Update README.md

securespider committed:
Update README.md

securespider committed:
Update README.md

securespider committed:
Removed unmappable character

Unmapped character causing issues for the ShadowJar file creation

Removed the unmappable character

securespider committed:
Updated gradle file based on requirements
